### PR TITLE
FE: Add Dialog component for #12

### LINKED ISSUE #12: FS: Authorization with Gmail API for the user
- Modify the user object to contain gmail authorization information
- If the user is not authorized when they open the web app, a material UI dialog (https://material-ui.com/components/dialogs/) if the user has not connected their account with their gmail, that pops up
- The auth with gmail button will open up gmail and redirect back the React front-end 
- The React front-end will send the auth token generated back

How to connect Gmail:
- https://developers.google.com/gmail/api/auth/about-auth
- https://developers.google.com/identity/protocols/OAuth2?hl=en_US
- Specific to web apps - https://developers.google.com/identity/protocols/OAuth2WebServer

**Task 1**:
- Method to get the authorization URL that you need to redirect the user to when they click the Google button on the front-end. I recommend having a route on the back-end to get the authorization url, and when someone clicks the button, you make an API call to that route and then you redirect the user manually to that route.

**Task 2**:
- Once someone fills out the authorization URL, they will be redirected back to your app. You need to then write another back-end API call to update get the user credentials information and then store it to the database.

I found this resource:
https://developers.google.com/gmail/api/auth/web-server
look at get_authorization_url - it actually has the code for step 1
look at exchange_code and get_credentials for step 2. In step 2, in the redirection URL, you will be passed a state and authorization code. You will have to write code on the front-end to send those detials to the back-end, and then in the back-end use some of the code in the resource to get the credentials, and then write an implementation to store the credentials in the back-end

### PR DESCRIPTION
**Pull request by cpascale43**
**What it does**
- Creates a Dialog component for use in Gmail authentication flow on Dashboard page ( see #12 ). 
- Adds dependency [react-google-button](https://www.npmjs.com/package/react-google-button). 

**What to look for**
- Try testing it by rendering it within a `/dialog` route in App.jsx.
- It may need to be modified later on for use within the Dashboard page. 

**What to avoid**
- Don't add any props, as dummy props are used. We'll be grabbing the actual props from UserContext later on.

### CODE REVIEW

File: client/.gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 .eslintrc.js
 .vscode/
 
-node_modules/
\ No newline at end of file
+node_modules/
+package-lock.json
```
Comment (skassam21) [inline client/.gitignore]: Generally actually don't need to gitignore package-lock.json (https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5)
It's also used by bots to determine if you have vulnerabilities, so it's good to keep

**Review by skassam21 — APPROVED** [review]
Looks good to me! Like the idea of the delay

File: client/.gitignore (hunk 1 above)
Comment (cpascale43) [reply client/.gitignore]: ok! updating now.

**Comment by mcgraths7** [comment]
Looks good to me! Can probably use portals to implement on the dashboard page when the time comes

**Pull request merged**